graph/arangodb/src/query.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{conversions, GraphArangoDbComponent, Transaction};
+use golem_graph::golem::graph::{
+    errors::GraphError,
+    query::{
+        Guest as QueryGuest, QueryExecutionResult, QueryOptions, QueryParameters, QueryResult,
+    },
+};
+
+impl Transaction {
+    pub fn execute_query(
+        &self,
+        query: String,
+        parameters: Option<QueryParameters>,
+        _options: Option<QueryOptions>,
+    ) -> Result<QueryExecutionResult, GraphError> {
+        let mut bind_vars = serde_json::Map::new();
+        if let Some(p) = parameters {
+            for (key, value) in p {
+                bind_vars.insert(key, conversions::to_arango_value(value)?);
+            }
+        }
+
+        let query_json = serde_json::json!({
+            "query": query,
+            "bindVars": bind_vars,
+        });
+
+        let response = self
+            .api
+            .execute_in_transaction(&self.transaction_id, query_json)?;
+
+        let result_array = response.as_array().ok_or_else(|| {
+            GraphError::InternalError("Expected array in AQL query response".to_string())
+        })?;
+
+        let query_result_value = if result_array.is_empty() {
+            QueryResult::Values(vec![])
+        } else {
+            let first_item = &result_array[0];
+            if first_item.is_object() {
+                let obj = first_item.as_object().unwrap();
+                if obj.contains_key("_id") && obj.contains_key("_from") && obj.contains_key("_to") {
```

Comment:
in all the clients: some parsing of "unstructured" data would be fine, but it is very hard to follow many of these mappings. ideally both these and error structures should use a proper serde mapping (for all clients)